File: client/node/authenticate.js
const {AuthenticateRequest} = require("./users_pb.js");

/**
 * @param {!proto.pb.Authentication.prototype} client The
 *     rpc client
 * @param body Payload authentication token
 * @param {!string} body.token Authentication token
 * @param callback
 *     call metadata
 *     callback The callback function(response)
 */
module.exports = function authenticate(client, body, callback) {
    const {
        token
    } = body;
    let request = new AuthenticateRequest();
    client.authenticate(request, {"authorization": token}, function(err, response) {
        if (err) {
            return callback({
                status: {
                    code: err.code,
                    details: err.message,
                    metadata: err.metadata
                },
                data: null
            });
        }
        return callback({
            status: {
                code: 0,
                details: "Authenticated successfully",
                metadata: response.metadata
            },
            data: {
                AuthId: response.getAuthid()
            }
        });
    });
}

File: client/go/authenticate.go
package client

import (
	"context"
	"github.com/tomashilaire/ct-users/client/go/pb"
	"google.golang.org/grpc/metadata"
	"log"
	"time"
)

// Authenticate calls sign up RPC
func (grpcClient *GrpcClient) Authenticate(token string) (string, error) {
	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
	defer cancel()

	newCtx := metadata.AppendToOutgoingContext(ctx, "authorization", token)
	resp, err := grpcClient.authService.Authenticate(newCtx, new(pb.AuthenticateRequest))
	if err != nil {
		log.Println("cannot authenticate, error: ", err)
		return "", err
	}
	return resp.AuthId, nil
}
